pkg/scheduler/core/generic_scheduler.go

```diff
@@ -1147,6 +1147,9 @@ func nodesWherePreemptionMightHelp(nodes []*v1.Node, failedPredicatesMap FailedP
 				unresolvableReasonExist = true
 				break
 			}
+			if unresolvableReasonExist {
```

Comment:
If unresolvableReasonExist is true, it would have broken out to line 1154; if unresolvableReasonExist is false, the change makes no sense.

So generally this change is invalid.

Reply:
@Huang-Wei If unresolvableReasonExist is true, it will break the "switch" block, but not break the "for" block.

Reply:
I see the pitfall. Then I'd suggest changing the whole section to:

(although LABEL is not a good pattern, but it's quite neat here)

```go
func nodesWherePreemptionMightHelp(nodes []*v1.Node, failedPredicatesMap FailedPredicateMap) []*v1.Node {
	potentialNodes := []*v1.Node{}
OUTERLOOP:
	for _, node := range nodes {
		failedPredicates, _ := failedPredicatesMap[node.Name]
		// If we assume that scheduler looks at all nodes and populates the failedPredicateMap
		// (which is the case today), the !found case should never happen, but we'd prefer
		// to rely less on such assumptions in the code when checking does not impose
		// significant overhead.
		// Also, we currently assume all failures returned by extender as resolvable.
		for _, failedPredicate := range failedPredicates {
			switch failedPredicate {
			case
				predicates.ErrNodeSelectorNotMatch,
				......
				predicates.ErrVolumeBindConflict:
				continue OUTERLOOP
			}
		}
		klog.V(3).Infof("Node %v is a potential node for preemption.", node.Name)
		potentialNodes = append(potentialNodes, node)
	}
	return potentialNodes
}
```

/cc @bsalamat for awareness.